frontend/src/query_builder/QueryVisualization.jsx

```diff
@@ -188,6 +188,11 @@ export default class QueryVisualization extends Component {
         }
     }
 
+    showDetailError() {
```

Comment:
I tweaked this code a little bit to follow what the current React guidelines are.  In this case it's fine to use `.getDOMNode()` though because the ref returns a valid instance of a component which has that method on it, so this is the expected way to access the DOM node after storing a reference to it.